protected/humhub/modules/admin/views/information/database.php

```diff
@@ -1,36 +1,64 @@
 <?php
+
 /**
  * @link https://www.humhub.org/
  * @copyright Copyright (c) 2017 HumHub GmbH & Co. KG
  * @license https://www.humhub.com/licences
  */
 
 use humhub\libs\Html;
+use humhub\modules\ui\view\components\View;
+use humhub\widgets\Pjax;
 
 /**
- * @var $this \humhub\modules\ui\view\components\View
+ * @var $this View
  * @var $databaseName string
- * @var $migrate string
+ * @var $linkRunMigrations string
  * @var $rebuildSearchRunning boolean
  */
 ?>
 <div>
     <p>
         <?php if ($rebuildSearchRunning): ?>
-            <div class="alert alert-info"><?= Yii::t('AdminModule.information', 'Search index rebuild in progress.'); ?></div>
+    <div class="alert alert-info"><?= Yii::t('AdminModule.information', 'Search index rebuild in progress.'); ?></div>
         <?php else: ?>
-            <?= Html::a('Rebuild search index', ['/admin/information/database', 'rebuildSearch' => 1], ['class' => 'btn btn-primary pull-right', 'data-method' => 'post', 'data-ui-loader' => '']); ?>
+            <?= Html::a(
+                'Rebuild search index',
+                ['/admin/information/database', 'rebuildSearch' => 1],
+                ['class' => 'btn btn-primary pull-right', 'data-method' => 'post', 'data-ui-loader' => '']
+            ); ?>
         <?php endif; ?>
 
-        <?= Yii::t('AdminModule.information', 'The current main HumHub database name is ') ?>
-        <i><b><?= Html::encode($databaseName) ?></b></i>
+    <?= Yii::t('AdminModule.information', 'The current main HumHub database name is ') ?>
+    <i><b><?= Html::encode($databaseName) ?></b></i>
     </p>
 </div>
 
 <p><?= Yii::t('AdminModule.information', 'Database migration results:'); ?></p>
 
-<div class="well">
-    <pre>
-        <?= $migrate; ?>
-    </pre>
+<div>
+    <?php
+    Pjax::begin([
```

Comment:
Do I understand correctly, that you want to get rid of the entire PJAX in this PR?

Reply:
Yes, currently we only use PJAX to avoid reloading of content container layouts or the whole page layout. And some users disabled this. (Don't know why). I don't want to introduce such a new approach like this by the way. 

For me, it is important that everything is implemented as identically as possible. Anything else makes hard to maintain.

I would like to discuss & decide this separately, if we should se PJAX in more places in the core/modules.

Reply:
> Yes, currently we only use PJAX to avoid reloading of content container layouts or the whole page layout. And some users disabled this. (Don't know why). I don't want to introduce such a new approach like this by the way.

Ok, I understand.

The reason why I chose PJAX was twofold:
1. Page loading speed/reactivity
2. Reducing the code loaded before migrations are happening

The first one is a general point. As a user I'd expect the page to move somehow once I click on a link. Here, nothing seems to happen (with PJAX enabled) until the request returns and the page is replaced. Hence, loading the page content quickly and then the migration (check) provides more responsiveness. However, this is just a detail in user experience.

The second point ties in with the topic of upgrading modules: The less code is loaded before the migration is happening, the less likely some code fails (e.g. because a field is not yet there). And the fact that some migration has not been a applied yet is an indication that something went wrong already before this point, as otherwise migrations would have been applied during installation/update.

As such, the PJAX sub-request only loads the controller and the view and runs the migration. There is no checking on any other part. And in an emergency, the link could also be called directly from the browser's address line (e.g. in a support case) to run the migration (if shell access isn't easy available to run it there). And as such some issues could be solved.

Now this just as a background for you. I'm happy to convert the code into plain page calls. Or, alternatively, in [JavaScript client](https://docs.humhub.org/docs/develop/javascript-client) calls. Would that be any better than PJAX? (Under the hood, PJAX is just hitchhiking the client xhr request. So I'm not sure if those who disable PJAX also disable any xhr.)

> For me, it is important that everything is implemented as identically as possible. Anything else makes hard to maintain.

Sure, got you.

Reply:
@martin-rueegg Thanks for the arguments and the changes. Perhaps we should really push for greater use of PJAX on the roadmap.